Feature profile listings integration #37

seyeonigbinde opened the pull request:
### What this PR does (required):
- Created the Profile Listing `Interface` in #37 
- Created the `mockProfileListing` file 
- Setup an `API` call for the Profile Listing
- Created a `useEffect` in the `useAuthContext` for the integration of the API
- Filtered the profileListing array for the handling of the search function in #37 
- The `ProfileList` and `SearchProfile` pages realigned in #58 

### Screenshots / Videos (front-end only):

<img width="1432" alt="Screenshot 2021-10-27 at 17 42 25" src="https://user-images.githubusercontent.com/79403650/139513677-21b7a902-6792-42c2-ab1f-a45e497bdba4.png">



### Any issues with the current functionality (optional):
- This integration has not been tested as the db has not been setup on AWS yet.

rajivtitus commented on client/src/context/useAuthContext.tsx:
```diff
@@ -29,6 +42,46 @@ export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
     [history],
   );
 
+  const getProfileContext = useCallback(
```
```suggestion
  const updateProfileContext = useCallback(
```

rajivtitus commented on client/src/context/useAuthContext.tsx:
```diff
@@ -29,6 +42,46 @@ export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
+            history.push('/profile-list');
+          } else {
+            setProfileList(undefined);
+            history.push('/dashboard');
```
```suggestion
```

We can eventually get rid of the dashboard as the listings page will serve as a home page for logged in users :)

rajivtitus commented on client/src/pages/Profile/ProfileList/SearchProfile.tsx:
```diff
@@ -1,54 +1,70 @@
+        </Grid>
       </Grid>
-    </Grid>
+      {filteredProfile.map((profiles: any, i: any) => (
```
Please avoid using 'any' as a type. You can use the 'Profile' interface here, and 'i' is a number

rajivtitus commented on client/src/pages/Profile/ProfileList/SearchProfile.tsx:
```diff
@@ -1,54 +1,70 @@
       </Grid>
-    </Grid>
+      {filteredProfile.map((profiles: any, i: any) => (
+        <ProfileList key={i} profiles={profiles} />
```
Usage of indexes as keys is in an anti-pattern in React, please consider using something unique instead

rajivtitus commented on client/src/pages/Profile/ProfileList/SearchProfile.tsx:
```diff
@@ -1,54 +1,70 @@
   const classes = useStyles();
+  const { profileList } = useAuth();
+
+  const filteredProfile = profileList.filter((profile: any) => {
```
Good work thinking about the search functionality. However, keep in mind that we need to implement the search functionality on the BE as it would be more efficient (let the DB do this work).

rajivtitus commented on client/src/pages/Profile/ProfileList/ProfileList.tsx:
```diff
@@ -1,56 +1,77 @@
+const initialValues = {
+  status: 'Lovely Dog Sitter',
+};
+export default function ProfileList({ profiles }: Props): JSX.Element {
```
```suggestion
export default function ProfileCard({ profiles }: Props): JSX.Element {
```

rajivtitus commented on client/src/pages/Profile/ProfileList/ProfileList.tsx:
```diff
@@ -1,56 +1,77 @@
+  const [value, setValue] = React.useState(3);
   const classes = useStyles();
 
+  const styles = {
```
Please add these styles to the useStyles file instead :)

rajivtitus commented on client/src/interface/Profile.ts:
```diff
@@ -9,3 +9,13 @@ export interface Profile {
   description: string;
 }
+
+export interface ProfileLists {
```
```suggestion
export interface SitterProfile {
```

I don't think this is a list of profiles

rajivtitus commented on client/src/helpers/APICalls/profileListing.ts:
```diff
@@ -0,0 +1,26 @@
+  const fetchOptions: FetchOptions = {
+    method: 'GET',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ firstName, lastName, address, description, availability, role, price }),
```
Since we are fetching all the sitter profiles, what is the reason for passing there args in the request.body?

seyeonigbinde replied:
We didn't create a separate Model for the Sitter Profiles, so I felt that it is better to destructure it. However, I will be deleting this off since I will be managing State within the sitter listings

rajivtitus commented on client/src/context/useAuthContext.tsx:
```diff
@@ -1,24 +1,37 @@
 
 export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
   const [loggedInUser, setLoggedInUser] = useState<User | null | undefined>();
+  const [profileList, setProfileList] = useState<ProfileLists | null | undefined>();
```
```suggestion
  const [profileList, setProfileList] = useState<SitterProfile[]>([]);
```

rajivtitus replied:
Good work trying to implement React's Context API. However, in this scenario it would be better to store sitter profiles within the listings page rather than a context (each time we make an API call to fetch sitter profiles based on a search condition it will cause the app to re-render multiple times)

rajivtitus commented on client/src/context/useAuthContext.tsx:
```diff
@@ -29,6 +42,46 @@ export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
+      role: 'sitter';
+      price: number;
+    }) => {
+      await profileListAPI(firstName, lastName, address, description, availability, role, price).then(
```
Same as comment below, why are we passing params to this API endpoint? :)

rajivtitus reviewed (commented):
Good start Seye! Please have a look at the comments and address them :)

seyeonigbinde commented:
Managed State now in the SearchProfile page instead of Context API #58 
Changed Interface from ProfileListing to SitterProfile in #37 
Transferred CarMedia style to useStyle file in #37 
Deleted MockProfileList from #37 
Deleted the ProfileListing from Helper/API call
@rajivtitus kindly review.
